src/main/java/ch/njol/skript/effects/EffOpenInventory.java

```diff
@@ -34,125 +36,201 @@
-			openFlag = parseResult.mark ^ 5;
-			invType = DISPENSER;
-		} else if(parseResult.mark >= 4) {
-			openFlag = parseResult.mark ^ 4;
-			invType = DROPPER;
-		} else if(parseResult.mark >= 3) {
-			openFlag = parseResult.mark ^ 3;
-			invType = HOPPER;
-		} else if (parseResult.mark >= 2) {
-			openFlag = parseResult.mark ^ 2;
-			invType = ANVIL;
-		} else if (parseResult.mark >= 1) {
-			openFlag = parseResult.mark ^ 1;
-			invType = CHEST;
-		} else if (parseResult.mark >= 0) {
-			invType = WORKBENCH;
-			openFlag = parseResult.mark ^ 0;
-		} else {
-			openFlag = parseResult.mark;
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (matchedPattern == 1) {
+			open = true;
+			assert !parseResult.tags.isEmpty() : "Syntax incorrectly handled parse result tags.";
+			syntax = OpenableInventorySyntax.valueOf(parseResult.tags.get(0).toUpperCase(Locale.ENGLISH));
+			if (!Skript.isRunningMinecraft(syntax.getVersion())) {
+				Skript.error("Opening an inventory of type '" + syntax.toString().toLowerCase(Locale.ENGLISH) + "' is only present on Minecraft version " + syntax.getVersion());
+				return false;
+			}
+			if (!syntax.doesMethodExist()) {
+				Skript.error(syntax.getMethodError());
+				return false;
+			}
 		}
-		
-		open = matchedPattern == 0;
-		invi = open ? exprs[0] : null;
+		inventory = matchedPattern == 0 ? exprs[0] : null;
 		players = (Expression<Player>) exprs[exprs.length - 1];
-		if (openFlag == 1 && invi != null) {
-			Skript.warning("Using 'show' inventory instead of 'open' is not recommended as it will eventually show an unmodifiable view of the inventory in the future.");
+		if (inventory instanceof Literal) {
+			Literal<?> literal = (Literal<?>) inventory;
+			Object object = literal.getSingle();
+			if (object instanceof InventoryType && !((InventoryType) object).isCreatable()) {
+				Skript.error("You can't open a '" + literal.toString() + "' inventory to players. It's not creatable.");
+				return false;
+			}
 		}
 		return true;
 	}
-	
+
 	@Override
-	protected void execute(final Event e) {
-		if (invi != null) {
-			Inventory i;
-			
-			assert invi != null;
-			Object o = invi.getSingle(e);
-			if (o instanceof Inventory) {
-				i = (Inventory) o;
-			} else if (o instanceof InventoryType) {
-				i = Bukkit.createInventory(null, (InventoryType) o);
-			} else {
-				return;
-			}
-			
-			if (i == null)
+	protected void execute(Event event) {
+		if (inventory != null) { // Show
+			Inventory inventory = null;
+			Object object = this.inventory.getSingle(event);
+			if (object == null)
 				return;
-			for (final Player p : players.getArray(e)) {
-				try {
-					p.openInventory(i);
-				} catch (IllegalArgumentException ex){
-					Skript.error("You can't open a " + i.getType().name().toLowerCase(Locale.ENGLISH).replaceAll("_", "") + " inventory to a player.");
+			for (Player player : players.getArray(event)) {
```

Comment:
There is a behavior change here. A unique inventory is now opened for each player. Is this intentional, and, if so, a change we want to make?

Reply:
DROPPER, DISPENSER and ANVIL already had this prior. The syntax is setup to create a unique inventory for each player. If the scripter wants to set a shared inventory, they use the %inventory/inventorytype% with `a new chest inventory` like everyone currently does. If the player is provided it should be added rather than null.